manual/manual/cmds/unified-options.etex

```diff
@@ -269,14 +269,39 @@ but before the standard library directory. See also option "-nostdlib".
 
 If the given directory starts with "+", it is taken relative to the
 standard library directory.  For instance, "-I +unix" adds the
-subdirectory "unix" of the standard library to the search path.
+subdirectory "unix" of the standard library to the load path.
+
+Note that "<dir>" will be searched whenever the compiler needs to look up a file
+given by a relative path. This includes the case of compilation objects of
+top-level modules mentioned in the program text, but also relative paths passed
+on the command-line.
 
 \top{%
 Directories can also be added to the list once
 the toplevel is running with the "#directory" directive
 (section~\ref{s:toplevel-directives}).
 }%top
 
+\item["-inc" \var{file}]
+
+Add the given file to the {em load path} (see also option "-I"). In contrast to
+"-I", this flag only adds the given file to the load path, the rest of the files
+in the parent directory of "file" remain hidden from the compiler unless they
+are added to the load path in some other way.
+
+As for "-I", files added to the load path using this flag are inserted in order
+after the current directory but before the standard library directory, and
+precedence is left-to-right.
+
+Similarly, if the given file starts with "+", it is taken relative to the
+standard library directory.  For instance, "-inc +list.cmi" adds the file
+"list.cmi" of the standard library to the load path.
+
+Note that in contrast to "-I", arguments to "-inc" will only be used when
+looking up paths consisting of a single basename. This includes the case of
+compilation objects of top-level modules mentioned in the program text, but also
+basenames passed on the command line.
```

Comment:
So the option `-inc foo/bar.cmo` will result in the query `bar.cmo` being resolved as `$(pwd)/foo/bar.cmo`, but the query `foo/bar.cmo` would be looked up in the rest of the load path?

Reply:
I don't understand the question, what is "a query"?

Reply:
Yes.

Reply:
> I don't understand the question, what is "a query"?

I believe @gasche refers to the act of resolving a filename using the load path (eg when the compiler seeks a `.cmi` file during compilation, or a `.cma` file during linking).

Reply:
Yes, I was confused by the fact that I couldn't see in which situation the compiler would be looking for "foo/bar.cmo".
But that would simply be when the file is given on the command line for linking.

So ignore me. :)

Reply:
> Yes.

This looks rather surprising. I would be in favor of a better semantics (if one exists), but at least (with the current semantics) we could point it out explicitly in the documentation?

Reply:
Just to make sure we are in the same page, when this was brought up in the past, you seemed to agree on this approach: https://github.com/ocaml/ocaml/pull/9056#issuecomment-640005038

Regarding the documentation, this is what the doc says (quoting from above):
> Note that in contrast to "-I", arguments to "-inc" will only be used when
looking up paths consisting of a single basename. This includes the case of
compilation objects of top-level modules mentioned in the program text, but also
basenames passed on the command line.

Reply:
I apologize with not being consistent with my past self (I understand that for the person doing the implementation work, shifting semantics requirements are a pain), but I have the impression that my message above was slightly confused. My initial intuition was that `-inc foo.cmi` would be reserved to in-language mention (and possibly "similar usages", but I don't know what other use-cases there are), and I failed at the time to notice or understand that your proposal would create a distinction, for path lookups in other contexst, between basename paths and non-basename paths.